### ISSUE TITLE
Shell Issue Tracker

### ISSUE THREAD

**Issue by sjmiles** [PolymerLabs/arcs#2262]
Tracking ticket for shell issues:

### p1

~~#2330 exceptions in firebase-storage.js (0_6_0)~~
~~#2065 Need Storage support for Arc serializations~~
~~#2097 Synthetic store sending malformed change events~~
~~#2105 HostedParticleShapes only resolve against context~~

### p1, not blocking transition to 0_6_0

#2089 SlotDomConsumer uses global templateCache
#2064 Storage `on('change')`: two `add` events, only one expected
#2121 serialization duplicates imports
#2104 Feature request: optional handle connections
#2120 duplicated imports cause duplication of stores

**Comment by shaper** [comment]
AIUI the issues here must be resolved for us to switch from the current shell to shell_0_6_0. Moving to the new shell is required for a wholly Pouch-based Pouch/Cloud-Planning scenario per @lindner @mmandlis.

**Comment by Cypher1** [comment]
@sjmiles can you explain on #2104 what you need so that you're unblocked?

**Comment by mmandlis** [comment]
@shaper afaik Cloud planning can work fine with older shell (@lindner are their any pouch related limitations?). It is just that there are some optimizations that I'm holding off until shell 0_6_0.

**Comment by sjmiles** [comment]
@shaper I made this to track all the shell-related bugs, they are not all blockers. The labels on the individual tickets should describe priority.

@mmandlis The older shell does not support Arcs-in-pouch, so the system is locked-in to firebase until 0_6_0.

@Cypher1 Thanks for looking at #2104. I added some more information there, I hope it clarifies. Also, this is not a blocker, I've marked it as p2.
